### PR TITLE
Update react teamplates to v.17

### PR DESCRIPTION
**Pull request by dgomezc**
## PR checklist

- Which issue does this PR relate to?
#1654 - Major update of react (17.*) and react-scripts (4.*)

### CODE REVIEW

File: templates/Web/Projects/ReactDefault/frontend/package.json [hunk 1]
```diff
@@ -3,18 +3,18 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
+    "@testing-library/jest-dom": "5.11.4",
```
Comment (prupipho) [inline templates/Web/Projects/ReactDefault/frontend/package.json:6]:
```suggestion
    "@testing-library/jest-dom": "5.11.6",
```

File: templates/Web/Projects/ReactDefault/frontend/package.json [hunk 2]
```diff
@@ -3,18 +3,18 @@
...
+    "@testing-library/react": "11.1.0",
```
Comment (prupipho) [inline templates/Web/Projects/ReactDefault/frontend/package.json:7]:
```suggestion
    "@testing-library/react": "11.2.2",
```

File: templates/Web/Projects/ReactDefault/frontend/package.json [hunk 3]
```diff
@@ -3,18 +3,18 @@
...
+    "@testing-library/user-event": "12.1.10",
```
Comment (prupipho) [inline templates/Web/Projects/ReactDefault/frontend/package.json:8]:
```suggestion
    "@testing-library/user-event": "12.5.0",
```

File: templates/Web/Projects/ReactDefault/frontend/package.json [hunk 4]
```diff
@@ -3,18 +3,18 @@
...
-    "react": "16.14.0",
-    "react-dom": "16.14.0",
+    "react": "17.0.1",
+    "react-dom": "17.0.1",
     "react-router-dom": "5.2.0",
-    "react-scripts": "3.4.4",
-    "fs-extra": "9.0.1"
+    "react-scripts": "4.0.1",
+    "fs-extra": "9.0.1",
+    "web-vitals": "1.0.1"
```
Comment (prupipho) [inline templates/Web/Projects/ReactDefault/frontend/package.json]:
```suggestion
    "web-vitals": "0.2.4"
```

**Comment by prupipho** [comment]
Do we want to update the App.test.js to use @testing-library/react as in the client?

```
import { render, screen } from '@testing-library/react';
import App from './App';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

```

File: templates/Web/Projects/ReactDefault/frontend/package.json (hunk 1 above)
Comment (prupipho) [reply templates/Web/Projects/ReactDefault/frontend/package.json:6]: Same as in angular. We will update at the end of the milestone before releasing.

**Review by prupipho — DISMISSED** [review]

**Review by prupipho — APPROVED** [review]

**Review by sibille — APPROVED** [review]

**Pull request merged**